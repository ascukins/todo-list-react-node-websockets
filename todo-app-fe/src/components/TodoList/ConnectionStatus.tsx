import React from 'react';
import styled from 'styled-components';

interface Props {
  connected: boolean;
}

export const ConnectionStatus: React.FC<Props> = ({ connected }) => {
  return (
    <>
      {connected ? (
        <Green>Connected to server.</Green>
      ) : (
        <Red>
          Not connected to server. Any local changes will be overwritten after
          connection.
        </Red>
      )}
    </>
  );
};

const Red = styled.div`
  color: red;
  margin-bottom: 30px;
  font-size: 22px;
  text-align: center;
`;

const Green = styled.div`
  color: green;
  margin-bottom: 30px;
  font-size: 22px;
  text-align: center;
`;
