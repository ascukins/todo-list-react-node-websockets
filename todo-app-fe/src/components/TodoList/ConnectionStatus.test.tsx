import { ConnectionStatus } from './ConnectionStatus';
import renderer from 'react-test-renderer';

// TODO do more tests for everything
test('ConnectionStatus connected', () => {
  const fragment = renderer
    .create(<ConnectionStatus connected={true} />)
    .toJSON();
  expect(fragment).toMatchSnapshot();
});

test('ConnectionStatus disconnected', () => {
  const fragment = renderer
    .create(<ConnectionStatus connected={false} />)
    .toJSON();
  expect(fragment).toMatchSnapshot();
});
