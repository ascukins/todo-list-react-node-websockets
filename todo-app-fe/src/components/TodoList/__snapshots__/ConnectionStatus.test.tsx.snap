// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus connected 1`] = `
<div
  className="sc-gsnTZi eJXUcq"
>
  Connected to server.
</div>
`;

exports[`ConnectionStatus disconnected 1`] = `
<div
  className="sc-bczRLJ kycCKY"
>
  Not connected to server. Any local changes will be overwritten after connection.
</div>
`;
